Writer.py

```diff
@@ -19,40 +19,35 @@ def read_from_file(self):
             return students
 
     def main(self):
-        self.students = self.read_from_file()
-        self.students.sort(key=lambda k: k['Searches'], reverse=True)
-        student = {}
-        while len(self.students) < self.size:
-            student = {'Name': input('enter name : '), 'Searches': int()}
-            self.students.append(student)
-            self.write_to_file(self.students)
-
-        new = input('list full,add new, enter yes or no ?')
-        if new == 'yes':
-            self.update()
-
-    def update(self):
-        students = self.read_from_file()
-        students.sort(key=lambda k: k['Searches'])
-        print('update this',students)
-        if len(students) == 0:
+        while True:
+            input_name = input('enter name: ')
+            self.students = self.read_from_file()
+            if len(self.students) < self.size:
```

Comment:
this should be <= (less than or equal to)

Reply:
thats done too